### PR TITLE
Action subfolder composite

### PR DESCRIPTION
**Pull request by ewels**
Alternative approach to https://github.com/nf-core/setup-nextflow/pull/174 where I've moved the node action into a subdirectory.

This almost certainly isn't going to work, just taking a punt.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -1,18 +1,36 @@
...
     default: "latest-stable"
   all:
-    description: "Whether to install every Nextflow release via the '-all' distribution."
+    description: "Whether to install every Nextflow release."
     required: false
     default: false
+  java-version:
+    description: "The Java version to use."
+    required: false
+    default: "17" # 21 suported as of Nextflow 24.05.0-edge. 11 is depreciated from 24.11.0-edge
```
Comment (mirpedrol) [inline action.yml]:
```suggestion
    default: "17" # 21 suported as of Nextflow 24.05.0-edge. 11 is deprecated from 24.11.0-edge
```

**Review by mirpedrol — APPROVED** [review]

**Comment by ewels** [comment]
Testing here: https://github.com/ewels/nf-core-rnaseq/pull/36

**Comment by ewels** [comment]
ok, this has been an epic all-day rabbit hole.

Nearly there now I think. To do:

- [ ] Strip out [`JasonEtco/build-and-tag-action`](https://github.com/JasonEtco/build-and-tag-action) from the `publish.yml` action and replicate the steps manually, using the composite action structure
    - I did this in the [`action-subfolder-composite-dist`](https://github.com/nf-core/setup-nextflow/tree/action-subfolder-composite-dist) branch manually for reference
- [ ] Probably eviscerate / delete `example.yml` workflow as I can't think how to get that to work on a relative path build
- [ ] Try releasing on a fork of this repo for testing purposes
- [ ] Release and hope it all works 🤞🏻 

Test with the manual `action-subfolder-composite-dist` branch seems to be working here: https://github.com/ewels/nf-core-rnaseq/pull/36 (there's a CI failure from nf-core lint but this is unrelated).

**Comment by MillironX** [comment]
> ok, this has been an epic all-day rabbit hole.
> 
> Nearly there now I think. To do:
> 
>     * [ ]  Strip out [`JasonEtco/build-and-tag-action`](https://github.com/JasonEtco/build-and-tag-action) from the `publish.yml` action and replicate the steps manually, using the composite action structure

:sweat_smile: Oh, yeah, the build and tag action has been broken for some time. It didn't like the change in output directory that Typescript brought. To test yourself, you will need to have a branch where you commit the `dist` directory (that mean going against gitignore, btw), and then you can run workflow tests. I just love JavaScript! :cry:

**Comment by MillironX** [comment]
Closing as #177 already took care of this

**Comment by ewels** [comment]
#177 missed a lot of what I found and fixed in this PR 😢 

Mostly: checking out the repo for the composite action. @mashehu now fighting the same problems and trying to hotfix.

**Pull request closed without merging**